File: spec/java/src/io/kaitai/struct/spec/TestParamsDef.java
package io.kaitai.struct.spec;

import io.kaitai.struct.ByteBufferKaitaiStream;
import io.kaitai.struct.KaitaiStream;
import io.kaitai.struct.testformats.ParamsDef;
import org.testng.annotations.Test;

import static org.testng.Assert.assertEquals;

public class TestParamsDef extends CommonSpec {
    @Test
    public void testParamsDef() throws Exception {
        KaitaiStream io = new ByteBufferKaitaiStream(SRC_DIR + "term_strz.bin");
        ParamsDef r = new ParamsDef(io, 5, true);

        assertEquals(r.buf(), "foo|b");
        assertEquals(r.trailer().intValue(), 0x61);
    }
}

File: spec/python/spec/test_params_def.py
import unittest
from testformats.params_def import ParamsDef
from kaitaistruct import KaitaiStream

class TestParamsDef(unittest.TestCase):
    def test_params_def(self):
        io = KaitaiStream(open('src/term_strz.bin', 'rb'))
        with ParamsDef(5, True, io, None, None) as r:
            self.assertEqual(r.buf, "foo|b")
            self.assertEqual(r.trailer, 0x61)
